fix(menu): menu-title-content style

coding-ice opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues
- fix https://github.com/ant-design/ant-design/issues/51361
- fix https://github.com/ant-design/ant-design/issues/51469
- fix #51526

### 💡 Background and Solution
在 5.21.0 中，上了 `extra` 属性，变更 `ant-menu-title-content` 为 `inline-flex`
- 导致这个元素的宽度从原来的 100% 变为子元素决定，引发了样式的 `break`

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 1. Fix css break caused by menu-title-content flex layout <br /> 2. Fixed an issue where the color of theme: 'dark' was not obvious          |
| 🇨🇳 Chinese | 1. 修复 menu-title-content：flex 布局 引发的 css break <br/> 2. 修复 theme: 'dark' 的颜色不明显问题 |

coding-ice committed:
fix(dropdown): dropdown menu title content style

coding-ice committed:
Merge branch 'master' into fix/extra

coding-ice committed:
chore: update rc-mentions to version 2.17.0 and rc-tabs to version 15.4.0

coding-ice committed:
Merge branch 'fix/extra' of ice:GetWebHB/ant-design into fix/extra

zombieJ commented on components/dropdown/style/index.ts:
```diff
@@ -242,10 +242,13 @@ const genBaseStyle: GenerateStyle<DropdownToken> = (token) => {
+            '&-extra': {
+              display: 'inline-flex',
+              alignItems: 'center',
+            },
```
demo 里加一条吧，这样好 image diff 看出来~

coding-ice committed:
test: add extra style debug

coding-ice committed:
test: snap

coding-ice committed:
test: snap

coding-ice committed:
Merge branch 'master' into fix/extra

coding-ice committed:
Refactor menu demo: remove extra-style-debug component and rename extra-style-debug files to extra-style

coding-ice committed:
Refactor menu demo: remove extra-style-debug component and rename extra-style-debug files to extra-style

coding-ice committed:
Refactor dropdown and menu styles: Add width to extra-style elements

coding-ice committed:
test: snap

coding-ice committed:
Merge branch 'master' into fix/extra

coding-ice committed:
Merge branch 'master' into fix/extra

coding-ice committed:
fix: Add Space component to menu demo

The Space component was missing from the import statement in the menu demo file. This caused a compilation error. The fix adds the Space component to the import statement, resolving the issue.

Refactor the menu demo to use Space component for vertical spacing between menus. This improves the visual layout and readability of the menus.

Fixes #51492

coding-ice committed:
test: snap

li-jia-nan commented:
这个现在什么进度啦？

coding-ice commented:
@zombieJ 麻烦再看下～

zombieJ commented on components/dropdown/style/index.ts:
```diff
@@ -242,10 +242,14 @@ const genBaseStyle: GenerateStyle<DropdownToken> = (token) => {
-            alignItems: 'center',
             flex: 'auto',
 
+            '&-extra': {
```
`-extra` 特指 extra 的 dom，不会做 layout。这里如果要加的话应该叫 `-with-extra`

coding-ice committed:
refactor: Update dropdown and menu title content class names

coding-ice committed:
Merge branch 'master' into fix/extra

afc163 committed:
Merge branch 'master' into fix/extra
